src/os/bluestore/BlueFS.cc

```diff
@@ -2229,6 +2240,227 @@ int64_t BlueFS::_read_random(
   return ret;
 }
 
+void BlueFS::_wal_update_size(FileRef file) {
+  uint64_t current_size = file->fnode.size;
+  uint64_t flush_offset = current_size;
+
+  FileReaderBuffer buf(super.block_size);
+
+  uint64_t x_off = 0;
+  uint64_t ret = 0;
+
+  std::unique_lock u_lock(file->lock);
+
+  while(flush_offset < file->fnode.allocated) {
+    // read first part of wal flush
+    {
+      buf.bl.clear();
+      buf.bl_off = p2align(flush_offset, (uint64_t)super.block_size);
+      auto p = file->fnode.seek(buf.bl_off, &x_off);
+      if (p == file->fnode.extents.end()) {
+        dout(5) << __func__ << " reading less then required "
```

Comment:
'than'

and - why not use fmt::format(), which is faster? (here and elsewhere in this patch)

Reply:
Do we use fmt here in ceph???? If I can get rid of stupid dout syntax for logs I would be the happiest of em all

Reply:
yes - we do